package-installer/src/main/java/com/vmware/pscoe/iac/installer/Installer.java

```diff
@@ -1380,68 +1385,124 @@ private static List<Package> getFilesystemPackages(final PackageType type) {
-					"Workflow execution failed: " + e.getClass().getName() + " : " + e.getLocalizedMessage(), e);
-		} catch (Exception e) {
-			throw new RuntimeException("General error during workflow execution: " + e.getClass().getName() + " : "
-					+ e.getLocalizedMessage(), e);
-		}
-	}
+    private static void runWorkflow(final Input input) throws ConfigurationException {
+        input.getText().getTextTerminal().println("Executing vRealize Orchestrator Workflow ...");
+
+        Gson gson = new GsonBuilder().setPrettyPrinting().serializeNulls().create();
+        String wfID = input.get(Option.VRO_RUN_WORKFLOW_ID);
+        String wfInputFilePath = input.get(Option.VRO_RUN_WORKFLOW_INPUT_FILE_PATH);
+        String wfOutputFilePath = input.get(Option.VRO_RUN_WORKFLOW_OUTPUT_FILE);
+        String wfErrFilePath = input.get(Option.VRO_RUN_WORKFLOW_ERR_FILE);
+        wfErrFilePath = StringUtils.isEmpty(wfErrFilePath) && !StringUtils.isEmpty(wfOutputFilePath)
+                ? wfOutputFilePath + ".err"
+                : "workflow.err";
+        String wfTimeout = input.get(Option.VRO_RUN_WORKFLOW_TIMEOUT);
+
+        Properties wfInput = new Properties();
+        try {
+            String wfInputString = new String(Files.readAllBytes(Paths.get(wfInputFilePath)));
+            if (Installer.isValidYaml(wfInputString)) {
+                wfInputString = Installer.convertYamlToJson(wfInputFilePath, wfInputString);
+            }
+
+            for (Map.Entry<String, JsonElement> entry : gson.fromJson(wfInputString, JsonObject.class).entrySet()) {
+                wfInput.put(entry.getKey(),StringEscapeUtils.escapeJava(gson.toJson(entry.getValue())));
+            }
+            List<String> prefixes = new ArrayList<>();
+            if (input.allTrue(Option.VRO_EMBEDDED)) {
+                prefixes.add(ConfigurationPrefix.VRO.getValue());
+                prefixes.add(ConfigurationPrefix.VRANG.getValue());
+            } else {
+                prefixes.add(ConfigurationPrefix.VRO.getValue());
+            }
+            RestClientVro client = RestClientFactory.getClientVro(ConfigurationVro.fromProperties(input.getMappings(prefixes.toArray(new String[0]))));
+            WorkflowExecution workflowExecutionResult = new VroWorkflowExecutor(client).executeWorkflow(wfID, wfInput, Integer.parseInt(wfTimeout));
+
+            JsonObject wfOutputJson = new JsonObject();
+            for (String key : workflowExecutionResult.getOutput().stringPropertyNames()) {
+                String value = StringEscapeUtils.unescapeJson(workflowExecutionResult.getOutput().getProperty(key));
+                wfOutputJson.add(key, gson.fromJson(value, JsonObject.class));
+            }
+            // write the workflow output
+            String outputString = gson.toJson(wfOutputJson);
+            if (Installer.endsWithIgnoreCase(wfOutputFilePath, "yaml")) {
+                outputString = Installer.convertJsonObjectToYaml(wfOutputFilePath, wfOutputJson);
```

Comment:
Was this also a requirement? I would rather we not do this, complicates for little to no benefit

Reply:
Discussed offline also with Dani, we agreed we want the output as well